src/http_utils.cpp

```diff
@@ -216,6 +218,22 @@ std::string http_utils::standardize_url(const std::string& url) {
     return result;
 }
 
+std::string http_utils::generate_random_upload_filename(std::string &directory) {
+    char *filename = NULL;
+    if (asprintf(&filename, "%s/%s", directory.c_str(), http_utils::upload_filename_template) == -1) {
+        return "";
+    }
+    int fd = mkstemp(filename);
```

Comment:
Not sure how portable mkstemp is (to be fair, the autobuild on windows will say that).

I think you might want to use a combination of `tpnam` and `fstream`. See: https://stackoverflow.com/a/7778959/756399